### PR TITLE
DEV-1494 RP subdomain magic link email

### PR DESCRIPTION
**Pull request by njharman**
#### What's this PR do?
Use Revenue Program specific subdomain for contributor portal magic link
email.

First look for RP subdomain in API POST, then from Referer header. If
not found in either, return 404.

Construct magic link from SITE_URL stripping off leaf subdomain and
preserving port it may hav

#### Why are we doing this? How does it help us?
Branding.

#### How should this be manually tested?
Setup RP subdomain and Request contributor link via email

#### Have automated unit tests been added?
Yes.

#### How should this change be communicated to end users?
N/A

#### Are there any smells or added technical debt to note?
There's a lot of subtle interdependencies between SITE_URL, what the SPA POSTs to API. I've added comments to hopefully head off surprises. 

The behavior of stripping off only leaf subdomain from SITE_URL may not be what is dessired.

Made comment about setting DOMAIN_APEX which seems to be manual version of "get base domain"

#### Has this been documented? If so, where?
Tickets, code.

#### What are the relevant tickets?
[DEV-1494]

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?
SITE_URL must be set and must include "scheme" (https://)

SPA must submit "subdomain" in POST /v1/contrib-email (or rely on Referer header, bad idea).  A 404 will be returned if missing.

#### Are there next steps? If so, what? Have tickets been opened for them?
SPA updated, yes.

[DEV-1494]: https://news-revenue-hub.atlassian.net/browse/DEV-1494?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by x110dc — APPROVED** [review]
LGTM.

**Comment by njharman** [comment]
The Build and Test Python failure "Checking for migrations" is confusing. I think its root cause is another commit 

`Your branch is up to date with 'origin/main'.
:~/.../news/rev-engine[rev-engine-wCuwcHw--py3.9](main*)$ ./manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, common, config, contenttypes, contributions, db, django_celery_beat, element_media, emails, organizations, pages, sessions, slack, thumbnail, users
Running migrations:
  No migrations to apply.
  Your models in app(s): 'pages' have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
  `
  
possibly this commit d001263d8 (Michael Ashton  2022-02-11 10:27:35 -0500 215)     revenue_program = models.ForeignKey("organizations.RevenueProgram", on_delete=models.CASCADE)

File: revengine/settings/base.py [hunk 1]
```diff
@@ -267,7 +267,16 @@
 ]
 
 
+# SITE_URL must include scheme and optionally port, https://example.com.
 SITE_URL = os.getenv("SITE_URL", "")
+# Application subdomains (that are NOT revenue program slugs)
+# TODO: shouldn't NON_D... include DenyListWords?
+NON_DONATION_PAGE_SUBDOMAINS = os.getenv("NON_DONATION_PAGE_SUBDOMAINS", "support:www").split(":")
+ORG_PORTAL_SUBDOMAINS = os.getenv("ORG_PORTAL_SUBDOMAINS", "support:").split(":")
```
Comment (x110dc) [inline revengine/settings/base.py:275]: These we both removed and subsumed by `DASHBOARD_SUBDOMAINS` in https://news-revenue-hub.atlassian.net/browse/DEV-1600 so they need to be removed here.

**Review by x110dc — CHANGES_REQUESTED** [review]
Minor change to remove two variables.

**Review by x110dc — APPROVED** [review]
LGTM.

**Pull request merged**